### PR TITLE
Cleanup `Cargo.toml` files & add taplo step in CI

### PR DESCRIPTION
**Pull request by hussein-aitlahcen**
- cleanup `dev-dependencies` of most pallets, they don't require `default-features = false`
- format `.toml` files using [taplo](https://taplo.tamasfe.dev/)
- introduce a new taplo step in CI linters

### CODE REVIEW

**Review by benluelo — DISMISSED** [review]

File: .taplo.toml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+include = ["Cargo.toml", "**/*.toml"]
+
+[formatting]
+# Align consecutive entries vertically.
+align_entries = true
```
Comment (cor) [inline .taplo.toml:5]: I'm not sure if I like this, as the reformating in order to preserve this property when adding new entries affects a lot of lines, making git diffs larger (so harder to see what changed)

Comment (benluelo) [reply .taplo.toml:5]: Personally, I think it's an OK trade-off for the readability benefits

Comment (hussein-aitlahcen) [reply .taplo.toml]: I think the giant diff is a bigger issue than having some part of the files aligned, I just used a default config and reverted it not to apply aligning.

**Review by cor — APPROVED** [review]
LGTM. Let's merge this, and if we observe that a formatting setting causes unwieldy diffs, then we can always revert that specific setting

**Review by benluelo — APPROVED** [review]

**Pull request merged**